### PR TITLE
Not try locking relations referencing to views

### LINKED ISSUE #5990: View is Not Distributed Error in Truncate with Cascade
``` SQL
CREATE TABLE tbl(a INT);
SELECT create_distributed_table('tbl', 'a');
CREATE VIEW vw AS SELECT * FROM tbl;

TRUNCATE TABLE tbl, vw CASCADE;

ERROR:  relation vw is not distributed
```

The error without distribution is different
``` SQL
CREATE TABLE tbl(a INT);
CREATE VIEW vw AS SELECT * FROM tbl;

TRUNCATE TABLE tbl, vw CASCADE;

ERROR:  "vw" is not a table
```

### PR DESCRIPTION
**Pull request by onurctirtir**
Since there can't be such a foreign key already.

This mainly fixes the error that Citus throws
when trying to truncate a distributed view.

Fixes #5990.

### CODE REVIEW

File: src/backend/distributed/utils/resource_lock.c [hunk 1]
```diff
@@ -1357,7 +1357,8 @@ AcquireDistributedLockOnRelations(List *relationList, LOCKMODE lockMode, uint32
 													 (void *) lockRelationRecord);
 		}
 
-		if ((configs & DIST_LOCK_REFERENCING_TABLES) > 0)
+		if (get_rel_relkind(relationId) == RELKIND_RELATION &&
```
Comment (onurctirtir) [inline src/backend/distributed/utils/resource_lock.c:1360]: Q: should we check for RELKIND_PARTITIONED_TABLE too, and maybe for RELKIND_FOREIGN_TABLE ?

Comment (gledis69) [reply src/backend/distributed/utils/resource_lock.c:1360]: I think you can switch to testing for relkind != RELKING_VIEW to cover all the non view cases.

**Review by gledis69 — COMMENTED** [review]
We might also want to check how the locking and truncating works with materialized views, but that might be outside of scope of this PR. For now, switching to != RELKIND_VIEW should fix the bug and cover all the cases the code was able to handle before.

**Review by gledis69 — CHANGES_REQUESTED** [review]
Blocking merging until we resolve the appropriate if check.
Candidates:
1) get_rel_relkind(relationId) != RELKIND_VIEW
2) get_rel_relkind(relationId) == RELKIND_RELATION || get_rel_relkind(relationId) == RELKIND_PARTITIONED_TABLE [|| get_rel_relkind(relationId) == RELKIND_FOREIGN_TABLE]

I think we can go with 1) since it is most straightforward. 
Some considerations about option 2:
This code path is only hit when doing truncate currently, meaning that checking for get_rel_relkind(relationId) == RELKIND_FOREIGN_TABLE is not useful since we can't truncate foreign tables. HOWEVER, as far as the function implementation is concerned, foreign tables can have foreign keys so I could argue adding get_rel_relkind(relationId) == RELKIND_FOREIGN_TABLE makes sense because that piece of code locks tables which reference the relation with foreign keys.

Both option 1 & 2 (with or without foreign table check) work as a solution.

File: src/backend/distributed/utils/resource_lock.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (onurctirtir) [reply src/backend/distributed/utils/resource_lock.c:1360]: Yeah, I was also thinking of the same,Will think about this little more on monday https://github.com/citusdata/citus/pull/6430#discussion_r995838732.

**Comment by onurctirtir** [comment]
> foreign tables can have foreign keys

Given that foreign tables cannot have a uniqueness constraint, no related can reference to them via a foreign key. So it seems that only checking for regular & partitioned tables is the correct approach here.

I deliberately prefer not checking for `!= RELKIND_VIEW` since is sounds safer to know which relkinds we're interested in.

**Review by gledis69 — APPROVED** [review]

**Pull request merged**